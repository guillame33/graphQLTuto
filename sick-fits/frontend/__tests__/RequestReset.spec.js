import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";

import RequestReset, {
  REQUEST_REST_MUTATION
} from "../components/RequestReset";

const mocks = [
  {
    request: {
      query: REQUEST_REST_MUTATION,
      variables: { email: "toto@toto.com" }
    },
    result: {
      data: {
        requestReset: { message: "success", __typename: "Message" }
      }
    }
  }
];

describe("<RequestReset />", () => {
  it("renders ans matches snapshot.", () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find("form[data-test='form']");
    expect(toJSON(form)).toMatchSnapshot();
  });
  it("calls the mutation", async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>
    );
    // simulate typing an email
    wrapper.find("input").simulate("change", {
      target: { name: "email", value: "toto@toto.com" }
    });
    wrapper.find("form").simulate("submit");
    await wait();
    wrapper.update();
    expect(wrapper.find("p").text()).toContain(
      "Check your email for a password link!"
    );
    // expect(toJSON(form)).toMatchSnapshot();
  });
});
