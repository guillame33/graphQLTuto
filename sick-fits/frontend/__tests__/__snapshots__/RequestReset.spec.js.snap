// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> renders ans matches snapshot. 1`] = `
<form
  className="Form-sc-1xszr8q-0 dMniLL"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Ask to reset your password
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset
    </button>
  </fieldset>
</form>
`;
